### PR TITLE
Add magic include variables

### PR DESCRIPTION
**Pull request by alexwhitman**
After the move to the AST based approach, it's now fairly trivial to add magic variables based on the current included file. This PR adds `__FILE` and `__DIR` which are set to the full include file path and the containing directory respectively.

These are analagous to the `__FILE__` and `__DIR__` magic constants in PHP (https://www.php.net/manual/en/language.constants.magic.php) and `__dirname` and `__filename` with Node.js (https://nodejs.org/dist/latest-v17.x/docs/api/globals.html#__dirname, https://nodejs.org/dist/latest-v17.x/docs/api/globals.html#__filename). There are likely other examples but these are two I'm familiar with.

This implements #49 requested by @adam12.

Opened as a draft for discussion. Things that may want consideration
- The variable naming.
- Is `__FILE` actually useful? `__DIR` is useful in an `include` context or other places where file paths are used.

### CODE REVIEW

**Comment by skx** [comment]
The implementation of this is nice and simple, so that's a positive thing.

The only thing I dislike is the naming of the variables, the underscore-prefix is horrible to look at!  But I can accept they're like node.js / PHP, and while I can't think of an immediate use for the file-part it wouldn't hurt to have both.

The only immediate use I considered was with a new `log` module:

```
log {  message => "I'm executing ${__FILE} on ${HOSTNAME}" }
```

That might not be a bad thing to allow, though I'd need to put together a simple log-module.  I'll open a pull-request for that this evening as it's independent.

In short, I'd be happy to merge as-is, though I don't love the underscore dropping them to just "FILE" and "DIR" would be almost worse since they're almost free of context.

**Comment by alexwhitman** [comment]
`INCLUDE_DIR` and `INCLUDE_FILE` could be better?

I can update the variable names and the README later today.

**Comment by skx** [comment]
I'd accept those names, thanks very much :)
